internal/resource/config/passwordcredentialvalidator/password_credential_validator_resource.go

```diff
@@ -122,6 +122,119 @@ func (r *passwordCredentialValidatorResource) Schema(ctx context.Context, req re
 	resp.Schema = schema
 }
 
+// Metadata returns the resource type name.
+func (r *passwordCredentialValidatorResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_password_credential_validator"
+}
+
+func (r *passwordCredentialValidatorResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func (r *passwordCredentialValidatorResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	var model passwordCredentialValidatorModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+
+	tableNames := []string{}
+	nestedTableFieldNameValues := []string{}
+	var rowCount int
+	configuration := model.Configuration.Attributes()
+	tables := configuration["tables"].(types.List).Elements()
+	for _, table := range tables {
+		tableAttrs := table.(types.Object).Attributes()
+		tableName := tableAttrs["name"].(types.String).ValueString()
+		tableNames = append(tableNames, tableName)
+		if tableName == "Users" {
+			tableRow := tableAttrs["rows"].(types.List).Elements()
+			rowCount = len(tableRow)
+			for _, row := range tableRow {
+				rowAttrs := row.(types.Object).Attributes()
+				fields := rowAttrs["fields"].(types.List).Elements()
+				for _, field := range fields {
+					fieldRow := field.(types.Object).Attributes()
+					nestedTableFieldName := fieldRow["name"].(types.String).ValueString()
+					nestedTableFieldNameValues = append(nestedTableFieldNameValues, nestedTableFieldName)
+				}
+			}
+		}
+	}
+
+	fieldNameValues := []string{}
```

Comment:
If all you ever do with these slices is check for existence of an element, I'd use a map[string]bool instead, so you don't have to iterate over the whole slice every time.